be/src/vec/functions/functions_geo.cpp

```diff
@@ -208,6 +208,42 @@ struct StDistanceSphere {
     }
 };
 
+struct StAngleSphere {
+    static constexpr auto NEED_CONTEXT = false;
+    static constexpr auto NAME = "st_angle_sphere";
+    static const size_t NUM_ARGS = 4;
+    static Status execute(Block& block, const ColumnNumbers& arguments, size_t result) {
+        DCHECK_EQ(arguments.size(), 4);
+        auto return_type = remove_nullable(block.get_data_type(result));
+
+        auto x_lng = block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        auto x_lat = block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        auto y_lng = block.get_by_position(arguments[2]).column->convert_to_full_column_if_const();
+        auto y_lat = block.get_by_position(arguments[3]).column->convert_to_full_column_if_const();
+
+        const auto size = x_lng->size();
+
+        MutableColumnPtr res = nullptr;
+        auto null_type = std::reinterpret_pointer_cast<const DataTypeNullable>(return_type);
+        res = ColumnNullable::create(return_type->create_column(), ColumnUInt8::create());
```

Comment:
You can use `block.get_data_type(result)` instead of `return_type` which is the nested type of `block.get_data_type(result)`

Reply:
done